src/lib/spinel/radio_spinel.cpp

```diff
@@ -1763,8 +1763,13 @@ void RadioSpinel::HandleTransmitDone(uint32_t          aCommand,
     }
 
 exit:
-    mState   = kStateTransmitDone;
-    mTxError = error;
+    // If current spinel frame is parsed failed, there should be some issues related to current transaction.
+    // At this time, we should raise a tx timeout failure then recover instead of setting the state to tx done.
+    if (error != OT_ERROR_PARSE)
```

Comment:
This is okay to add

But the question for me is how/why do we get a parse error here? 
- Is the RCP sending malformed spinel frame here? If so, we should figure it out and fix this on RCP?
- Is the bus used for communication between RCP/host is unreliable and can add errors? If so, should we add some error checking at BUS/interface layer?
- Any other possibilities?.

To me this situation seems be unlikely?

Reply:
Hi Abtink, as mentioned in the description, this is a corner issue based on my performance test (This parsing error occurred only twice during an approximately 83-hour test). To understand the reasons behind this parsing error, I need to conduct further investigation. However, I believe that considering this corner scenario is important if the code remains as it is now.

Reply:
> we should raise a tx timeout failure then recover instead of setting the state to tx don

@zwx1995esp  how about directly calling `HandleRcpTimeout()`  here instead of waiting for tx timeout to kick in from `ProcessRadioStateMachine()`?

Reply:
Hi, Abtink, that's also a reasonable process when the tx done frame has parsing error, I will try like this ASAP.

Reply:
Hi @abtink already changed like your suggestion and verified on my test env. PTAL.

Reply:
> To me this situation seems be unlikely?

I'm seeing this occur daily on around hundreds of devices in a test population. Applying the patch mitigates failures we're seeing around otbr crashing, but I'd like to better understand what's going on.

> Is the RCP sending malformed spinel frame here? If so, we should figure it out and fix this on RCP?

Abtin, I tagged you in a related issue if you would like me to try anything. I haven't figured out exactly what is causing this, but it seems to happen anywhere from 1-5 times a day.